airflow/jobs/scheduler_job.py

```diff
@@ -1512,18 +1545,42 @@ def _schedule_dag_run(self, dag_run: DagRun, session: Session) -> int:
         # TODO[HA]: Run verify_integrity, but only if the serialized_dag has changed
 
         # TODO[HA]: Rename update_state -> schedule_dag_run, ?? something else?
-        schedulable_tis = dag_run.update_state(session=session)
+        schedulable_tis = dag_run.update_state(session=session, handle_callback=False)
         # TODO[HA]: Don't return, update these from in update_state?
         count = session.query(TI).filter(
             TI.dag_id == dag_run.dag_id,
             TI.execution_date == dag_run.execution_date,
             TI.task_id.in_(ti.task_id for ti in schedulable_tis)
         ).update({TI.state: State.SCHEDULED}, synchronize_session=False)
 
-        # TODO[HA]: Manage SLAs
-
         return count
 
+    def _send_dag_callbacks_to_processor(self, dag_run: DagRun):
+        if not self.processor_agent:
+            raise ValueError("Processor agent is not started.")
+
+        dag = dag_run.get_dag()
+        self._manage_slas(dag)
+        if dag_run.callback:
+            self.processor_agent.send_callback_to_execute(dag_run.callback)
+
+    def _manage_slas(self, dag: DAG):
+        check_slas: bool = conf.getboolean('core', 'CHECK_SLAS', fallback=True)
```

Comment:
Make this a constant somewhere - class level or in `settings` module -- getboolean is relatively expensive and we don't want to check this every time.

Reply:
done in https://github.com/astronomer/airflow/pull/913/commits/f0d64d44e82eef4cb4a5dbd85969a44de3939b99